src/pages/spawn/spawnHost/fields/HostExpirationField.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Body } from "@leafygreen-ui/typography";
+import { set } from "date-fns";
+import DatePicker from "components/DatePicker";
+import { InputLabel } from "components/styles";
+import TimePicker from "components/TimePicker";
+
+export interface ExpirationDateType {
+  expiration: Date;
+  noExpiration: boolean;
+}
+
+interface HostExpirationFieldProps {
+  data: ExpirationDateType;
+  onChange: React.Dispatch<React.SetStateAction<any>>;
+}
+export const HostExpirationField: React.FC<HostExpirationFieldProps> = ({
+  onChange,
+  data,
+}) => {
+  const { expiration, noExpiration } = data;
+
+  const updateDate = (_, dateString) => {
```

Comment:
For these, I think the first parameter is the actual date object right? So wouldn't it be cleaner to use that and combine with time rather than parsing the string?

Reply:
Thanks good suggestion. not sure why we missed it last time.